Bluetooth: usb: bt_hci: enable ISO packet transfer

dsseng opened the pull request:
Allow ISO packets over USB, modeled after Linux btusb and hci_core.

Fixes #44013

dsseng commented on subsys/usb/device_next/class/bt_hci.c:
```diff
@@ -326,14 +434,34 @@ static int bt_hci_acl_out_cb(struct usbd_class_data *const c_data,
+
+		uint16_t handle = bt_acl_handle(sys_le16_to_cpu(hdr->handle));
+
+		// TODO: snoop ISO handles from ISO commands
+		// this would allow to use controllers other than ll_sw
+		if (IS_CIS_HANDLE(handle) || 
+			IS_ADV_ISO_HANDLE(handle) ||
+		    IS_SYNC_ISO_HANDLE(handle)) {
+			LOG_DBG("CIS/ISO handle %u, type %u", hdr->handle, type);
+			type = BT_HCI_H4_ISO;
+		}
```
Basically, I'm talking about this: https://github.com/torvalds/linux/blob/89be9a83ccf1f88522317ce02f854f30d6115c41/net/bluetooth/hci_core.c#L2939

We should keep track of CIS/BIS/other ISO setup/destroy commands host issues, and this would allow us to know when an ACL packet host sends us is an ISO packet.

cvinayak commented on subsys/usb/device_next/class/bt_hci.c:
```diff
@@ -35,6 +35,113 @@
 #include <zephyr/drivers/bluetooth.h>
 #include <zephyr/sys/atomic.h>
 
+// FIXME: HACK
```
NACK... please no specific Controller implementation to be referenced in the USB subsystem.

dsseng replied:
This is a hack I use while I work on other parts. Have you read my idea to get notion of which handles are ISO by capturing HCI commands? Do you think there might be a better idea?

cvinayak commented on samples/bluetooth/iso_central/src/main.c:
```diff
@@ -112,7 +112,7 @@ static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
 
 	/* connect only to devices in close proximity */
-	if (rssi < -50) {
+	if (rssi < -70) {
```
Revert back... change is not relevant to the scope of this PR.

```suggestion
	if (rssi < -50) {
```

cvinayak commented on samples/bluetooth/hci_usb/usbd_next_prj.conf:
```diff
@@ -15,3 +15,143 @@ CONFIG_USBD_BT_HCI=y
+CONFIG_BT_ISO_RX_BUF_COUNT=1
+
+# Controller
+CONFIG_BT_LL_SW_SPLIT=y
```
Again, no specific Controller implementation to be referenced in main prj.conf file.

dsseng committed:
Bluetooth: hci: clarify invalid ISO length error

Add the failed comparison to the message.

Signed-off-by: Dmitrii Sharshakov <d3dx12.xx@gmail.com>

dsseng committed:
Bluetooth: usb: bt_hci: enable ISO packet transfer

Allow ISO packets over USB, modeled after Linux btusb
and hci_core mechanisms.

Signed-off-by: Dmitrii Sharshakov <d3dx12.xx@gmail.com>

dsseng commented on subsys/usb/device_next/class/bt_hci.c:
```diff
@@ -218,6 +325,7 @@ static void bt_hci_tx_thread(void *p1, void *p2, void *p3)
 			ep = bt_hci_get_int_in(c_data);
 			break;
 		case BT_HCI_H4_ACL:
+		case BT_HCI_H4_ISO:
 			ep = bt_hci_get_bulk_in(c_data);
```
So far I couldn't make Linux host recognize ISO packets. It might be the case it ignores CIS creation commands if passed via userchan. If that's the case, userchan driver in Zephyr might have to do this instead

jfischer-no reviewed (changes requested):
ISO support relies on the parts of the Bluetooth spec that is not public and are only available to the BT SIG members. There are patches in private repositories for both sides that cannot be public yet. This PR does not make sense and should be closed.
